src/async_impl/client.rs

```diff
@@ -1374,13 +1390,24 @@ impl ClientBuilder {
                 return self;
             }
         }
-        #[cfg(feature = "__rustls")]
+        #[cfg(all(feature = "__rustls", not(feature = "http3")))]
         {
             if let Some(conn) =
                 (&mut tls as &mut dyn Any).downcast_mut::<Option<rustls::ClientConfig>>()
             {
-                let tls = conn.take().expect("is definitely Some");
-                let tls = crate::tls::TlsBackend::BuiltRustls(tls);
+                let rustls = conn.take().expect("is definitely Some");
+                let tls = crate::tls::TlsBackend::BuiltRustls { rustls };
+                self.config.tls = tls;
+                return self;
+            }
+        }
+        #[cfg(feature = "http3")]
+        {
+            if let Some(conn) = (&mut tls as &mut dyn Any)
+                .downcast_mut::<Option<(rustls::ClientConfig, quinn::TransportConfig)>>()
```

Comment:
I don't think I'd want to do this, because I would like to make it easier to swap in s2n-quic or similar for the QUIC implementation. Is there need to configure the `TransportConfig` yourself, or can we just figure it out internally?

Reply:
> I don't think I'd want to do this, because I would like to make it easier to swap in s2n-quic or similar for the QUIC implementation.

How does this prevent swapping in `s2n-quic`?

> Is there need to configure the `TransportConfig` yourself, or can we just figure it out internally?

There's a bunch of configuration options that aren't exposed through `ClientBuilder`.
Personally I don't need it and we can definitely figure it out internally.

I will adjust the PR to generate `TransportConfig` instead of having to pass it through.

Reply:
Done.